backend/jsonProcessor.js

```diff
@@ -0,0 +1,132 @@
+      console.error("Error reading JSON file:", err);
+      return;
+    }
+    try {
+      const jsonData = JSON.parse(data);
+
+      for (const business of jsonData.results) {
+        const location = `${business.geometry.location.lat}, ${business.geometry.location.lng}`;
+        const businessType = business.types.join(", ");
+        const firstType = business.types[0]
+          ? business.types[0].trim()
+          : "Uncategorized";
+        const businessHours = business.opening_hours
+          ? business.opening_hours.weekday_text || ["Unknown"]
+          : ["Unknown"];
+
+        // Fetch additional details from Google Places API
+        const placeDetails = await fetchPlaceDetails(business.place_id);
+
+        const upsertedBusiness = await prisma.business.upsert({
+          where: { placeId: business.place_id },
+          update: {
+            name: placeDetails.name || business.name,
+            location,
+            contactInformation:
+              placeDetails.formatted_phone_number || "Unknown",
+            overview: placeDetails.editorial_summary
+              ? placeDetails.editorial_summary.overview
+              : "Overview not provided",
+            businessHours: placeDetails.current_opening_hours
+              ? placeDetails.current_opening_hours.weekday_text
+              : [],
+            averageRating: placeDetails.rating || business.rating,
+            businessType,
+            photoReference: placeDetails.photos
+              ? placeDetails.photos[0].photo_reference
+              : null,
+            photos: placeDetails.photos
+              ? placeDetails.photos.map((photo) => photo.photo_reference)
+              : [],
+            category: firstType,
+          },
+          create: {
+            placeId: business.place_id,
+            name: placeDetails.name || business.name,
+            location,
+            contactInformation:
+              placeDetails.formatted_phone_number || "Unknown",
+            overview: placeDetails.editorial_summary
+              ? placeDetails.editorial_summary.overview
+              : "Overview not provided",
+            businessHours: placeDetails.current_opening_hours
+              ? placeDetails.current_opening_hours.weekday_text
+              : [],
+            averageRating: placeDetails.rating || business.rating,
+            businessType,
+            photoReference: placeDetails.photos
+              ? placeDetails.photos[0].photo_reference
+              : null,
+            photos: placeDetails.photos
+              ? placeDetails.photos.map((photo) => photo.photo_reference)
+              : [],
+            category: firstType,
+            recommendations: { create: [] },
+          },
+        });
+
+        if (placeDetails.reviews) {
+          for (const review of placeDetails.reviews) {
+            const existingReview = await prisma.review.findFirst({
+              where: {
+                name: review.author_name,
```

Comment:
What are you trying to achieve when there is already an existing review? It looks like we're updating the profile photo, but I'm unsure since we already do this on first pass with prisma.review.create.

Reply:
When a new review is fetched from the Google Places API, we first check if a review by the same author already exists for the given business. If no existing review is found, we create a new review using prisma.review.create.
If an existing review is found, we update the profilePhoto field using prisma.review.update. This ensures that if a user updates their profile picture, our data reflects this change.